src/backend/distributed/metadata/metadata_cache.c

```diff
@@ -691,63 +706,81 @@ ShardPlacementList(uint64 shardId)
 
 
 /*
- * LookupShardIdIndexSlot returns the hash entry mapping shard id to index.
- * The returned slot's shardId will be 0 when not found.
- * While initializing the caller will set shardId to the searched value,
- * while searching the caller will ignore the slot.
- * There should always be spare capacity; this fails to halt otherwise.
+ * LookupShardCacheEntry returns the cache entry belonging to a shard, or
+ * errors out if that shard is unknown.
  */
-static ShardIdIndexSlot *
-LookupShardIdIndexSlot(ShardIdIndexSlot *table, size_t tableSize, uint64 shardId)
+static ShardIdCacheEntry *
+LookupShardIdCacheEntry(int64 shardId)
```

Comment:
This function is mostly a copy of LookupShardCacheEntry as it existed before #3866.